Source/WebCore/inspector/InspectorCanvas.cpp

```diff
@@ -610,35 +610,35 @@ std::optional<InspectorCanvasCallTracer::ProcessedArgument> InspectorCanvas::pro
 {
     if (!argument)
         return std::nullopt;
-    return {{ JSON::Value::create(0), RecordingSwizzleType::WebGLBuffer }};
+    return {{ JSON::Value::create(static_cast<int>(argument->object())), RecordingSwizzleType::WebGLBuffer }};
 }
 
 std::optional<InspectorCanvasCallTracer::ProcessedArgument> InspectorCanvas::processArgument(WebGLFramebuffer* argument)
 {
     if (!argument)
         return std::nullopt;
-    return {{ JSON::Value::create(0), RecordingSwizzleType::WebGLFramebuffer }};
+    return {{ JSON::Value::create(static_cast<int>(argument->object())), RecordingSwizzleType::WebGLFramebuffer }};
```

Comment:
Is this number something that is usable by other JS APIs?  Or is it purely internal state?

If the latter, I don't think we want this to be something we expose as there's not really a benefit to the developer.  If your goal is to just have some sort of way to differentiate between different instances of the same object (e.g. "Framebuffer #1" vs "Framebuffer #2"), then I'd argue for having a `HashMap` of identifiers instead of using the underlying `GCGLuint`.

 It's also probably gonna give you grief when dumping the recording JSON if these numbers are not deterministic.

Reply:
Generally they are not usable, but I think it's ok to expose them.

Reply:
maybe we should have it match how we show internal identifiers for heap snapshots? <https://webkit.org/blog/6425/memory-debugging-with-web-inspector/>

Reply:
My goal is to have a way to work out which WebGL object is being referred to by WebGL calls in the trace. Currently, I see `glUseProgram(WebGLProgram)` which is completely useless when using the Web Inspector to try to diagnose rendering issues. (Similarly, there's no buffer data, so I don't know what data is being passed to draw calls)

OpenGL names all objects via integer ids. They're completely free-form, but there are OpenGL calls that generate them. OpenGL developers are used to using these ids to connect calls in traces.

What's the difference between displaying "Framebuffer #1" and "Framebuffer 15"? How would I store the `HashMap` of identifiers in the `InspectorCanvasCallTracer`?

Reply:
I absolutely agree that we should display something for these.

The problem with this approach, however, is that it doesn't really help the developer understand "which buffer is this".  Just because WebKit happens to call this "Framebuffer #2" doesn't mean the developer will know which buffer that actually is.  Sure, it helps distinguish two calls to the same function that both use buffers, but not really much more than that.

If we really wanted to do that we'd probably need to keep track of the spot where the buffer is created (e.g. have the UI indicate the identifier of the buffer next to the command that created it, something like `createBuffer() -> (@1)`).  For that, you'll probably want to either adjust the IDL generator to capture the return value from things that are instrumented using a `CallTracer` or add specific `InspectorInstrumentation` hooks in the functions that create these objects (e.g. `WebGLRenderingContextBase::createBuffer`).

As far as having a `HashMap`, you'd probably want that to be on `InspectorCanvas` alongside some of the other things we already have there (e.g. `DuplicateDataVariant`).

Reply:
Yeah, we definitely need to have `createBuffer` tell the user what the identifier is, but I think that can come later. For now it's still pretty helpful to know that parameters are different.